openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -110,7 +112,8 @@ def update_user_preferences(requesting_user, update, username=None):
         PreferenceUpdateError: the operation failed when performing the update.
         UserAPIInternalError: the operation failed due to an unexpected error.
     """
-    existing_user = _get_user(requesting_user, username)
+    if not existing_user:
```

Comment:
@cahrens @jcdyer 
How about we separate out authorization code from `_get_user` and make it a separate helper function called from `_get_user` as well. This way we can update
```
def update_user_preferences(requesting_user, update, user=None): # user can either be username string or user object
    if user and not isinstance(user, str):
        _is_authorized(requesting_user, user.username)
    else:
        existing_user = _get_user(requesting_user, user)  #here user is the username
```